app/services/support_interface/notes_export.rb

```diff
@@ -0,0 +1,32 @@
+module SupportInterface
+  class NotesExport
+    LABELS = %w[
+      note_subject note_message note_created_at candidate_id provider_code provider_name provider_user_id
+      number_of_training_provider_relationships total_number_of_provider_relationships
+    ].freeze
+
+    def data_for_export
+      data = Note.joins(
+        [:provider_user, application_choice: [{ application_form: :candidate }, { course_option: { course: :provider } }]],
+      ).joins(
+        'LEFT JOIN provider_relationship_permissions tpp ON tpp.training_provider_id = providers.id',
+      ).joins(
+        'LEFT JOIN provider_relationship_permissions rpp ON rpp.ratifying_provider_id = providers.id',
+      ).group(
+        :subject, :message, :created_at, :"candidates.id", :"providers.code", :"providers.name", :"provider_users.id"
+      ).pluck(
+        :subject,
+        :message,
+        :created_at,
+        :"candidates.id",
+        :"providers.code",
+        :"providers.name",
```

Comment:
I suspect the intention of the column here is to capture which provider a user is from when making the note. I guess that means we do need to include the offered_option too. Worth checking though

Reply:
I'll check this, the value can be multiple provider names and codes if coming from the join to the provider user.

Reply:
The desire here is to export the course provider assuming it's a provider belonging to the provider user making the note, otherwise we can safely assume the note was made by someone on the accredited provider side and export that provider instead.